### ISSUE TITLE
Missing case in `prefer-switch`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1393]
```js
if (foo === '1') {
  ...
}
if (foo === '2') {
  ...
}
if (foo === '3') {
  ...
}
if (foo === '4') {
  ...
}
```

To fix this case, we can simply only check if `foo` is declared as `const`.

This come from my work, but real case is more complicated, it's like

```js
if (foo.bar === '1') {
  ...
}
if (foo.bar === '2') {
  ...
}
if (foo.bar === '3') {
  ...
}
if (foo.bar === '4') {
  ...
}
```

To fix this case, we need to check `foo.bar` and `foo` didn't get changed in each `if` block, I feel this is hard to do.

**Comment by sindresorhus** [comment]
> we need to check foo.bar and foo didn't get changed in each if block, I feel this is hard (even impossible) to do.

Another case where type info would have made this easy...

**Comment by fisker** [comment]
Or maybe we can just report it without fix. It's not common to change `foo.bar` in one `if` to let another `if` to match, especially with many `if`s on the same object.

**Comment by sindresorhus** [comment]
Yeah, let's do that.

**Comment by fisker** [comment]
I changed my mind about this, I think this should be a separate rule, maybe called `prefer-else-if`, `prefer-switch` has a limitation on `minimumCases`,

```js
if (foo === '1') {
  ...
}
if (foo === '2') {
  ...
}
```

There are only two cases, we should not ask user to use `switch`, but


```js
if (foo === '1') {
  ...
} else if (foo === '2') {
  ...
}
```

should still be preferred, because the old one always compares `foo` twice, event value of `foo` is `1`.

After we fix it to `else-if`, if there are many cases,  `prefer-switch` will catch it.

**Comment by sindresorhus** [comment]
Sounds good.
